modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -419,15 +428,60 @@ def take_2d_labels_or_positional(
         if row_labels is not None:
             raise NotImplementedError("Row labels masking is not yet supported")
 
-        if row_positions is not None:
-            base = base._maybe_materialize_rowid()
-            op = MaskNode(base, row_labels=row_labels, row_positions=row_positions)
-            return self.__constructor__(
-                columns=base.columns,
-                dtypes=base.copy_dtypes_cache(),
-                op=op,
-                index_cols=base._index_cols,
-                force_execution_mode=base._force_execution_mode,
+        if row_positions is None:
+            return base
+
+        # Check if row_positions could be converted to a range.
+        if len(row_positions) > 2 and not is_range_like(row_positions):
+            diff = row_positions[1] - row_positions[0]
+            is_range = True
+            for i in range(2, len(row_positions)):
+                if (row_positions[i] - row_positions[i - 1]) != diff:
+                    is_range = False
+                    break
+            if is_range:
+                row_positions = range(row_positions[0], row_positions[-1] + diff, diff)
+
+        base = base._maybe_materialize_rowid()
+        op = MaskNode(base, row_labels=row_labels, row_positions=row_positions)
+        base = self.__constructor__(
+            columns=base.columns,
+            dtypes=base.copy_dtypes_cache(),
+            op=op,
+            index_cols=base._index_cols,
+            force_execution_mode=base._force_execution_mode,
+        )
+
+        # Reverse the frame rows, if performing a reverse order selection via HDK.
```

Comment:
Do I and correctly that we sacrifice performance to get this particular case work?

Reply:
Yes, but it seems the only way to reverse rows via HDK.

Reply:
Got it, thanks. ANyway it looks like a subject for submitting a bug in hdk. This can be done later of course

Reply:
Does it mean we also completely lose an option to arrow-execute this case?

Reply:
No, below is the condition - `and not base._can_execute_arrow()`. If we have an arrow table in the partition, it will be executed with arrow.